infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -564,8 +564,9 @@ private void updateTopologyConfigId(long configId) {
      */
     public void processLockRelease() {
         log.debug("Lock released");
-        isLeader.set(false);
+        // Unset isLeader flag after stopping log replication
```

Comment:
Nan can you explain this a bit more, how changing the order fixes a bug, can we write a test?

Reply:
> Nan can you explain this a bit more, how changing the order fixes a bug, can we write a test?

function `stopLogReplication()` has an if condition about isLeader flag. 


```
    /**
     * Stop Log Replication
     */
    private void stopLogReplication() {
        if (localClusterDescriptor != null && localClusterDescriptor.getRole() == ClusterRole.ACTIVE && isLeader.get()) {
            log.info("Stopping log replication.");
            replicationManager.stop();
        }
    }
```


However, `isLeader` flag was unset before calling it, which means it will never be true in this lock release case.